File: libraries/hds/cmp_type.f
      subroutine cmp_type(struct, comp, type, status)
*+
*  Name:
*     CMP_TYPE

*  Purpose:
*     Component Type enquiry.

*  Language:
*     VAX Fortran

*  Invocation:
*     CALL CMP_TYPE(LOC, NAME, TYPE, STATUS)

*  Description:
*     A type enquiry is made for a structure component.

*  Arguments:
*     LOC=CHARACTER*(DAT__SZLOC)
*        Variable containing a locator associated with a structured
*        data object.
*     NAME=CHARACTER*(*)
*        Expression specifying the component name of a primitive
*        object contained in the structure.
*     TYPE=CHARACTER*(*)
*        Variable to receive the data type.
*        It must be of sufficient precision to contain the type
*        string.
*     STATUS=INTEGER
*        Variable holding the status value. If this variable is not
*        SAI__OK on input, the routine will return without action.
*        If the routine fails to complete, this variable will be
*        set to an appropriate error number.

*  Algorithm:
*     Get object locator using DAT_FIND and then get type from
*     DAT_TYPE.

*  Authors:
*     Jack Giddings (UCL::JRG)
*     {enter_new_authors_here}

*  History:
*     3-JAN-1983:  Original.  (UCL::JRG)
*     15-APR-1987:  Improved prologue layout (RAL::AJC)
*     {enter_further_changes_here}

*  Bugs:
*     {note_any_bugs_here}

*-

*  Global Constants:
      INCLUDE 'SAE_PAR'
      INCLUDE 'DAT_PAR'

*  Arguments Given:
      character*(*) struct		! Structure Locator
      character*(*) comp		! Component Name

*  Arguments Returned:
      character*(*) type		! Component type
*    Status return :
      integer status			! Status Return

*  Local Variables:
      character*(DAT__SZLOC) loc	! Component locator

*.


      if (status .eq. SAI__OK) then
         call dat_find(struct, comp, loc, status)
         if (status .ne. SAI__OK) then
            call cmp_erdsn(struct, comp, status)
         else
            call dat_type(loc, type, status)
            if (status .ne. SAI__OK) then
               call cmp_erdsn(struct, comp, status)
            endif
            call dat_annul(loc, status)
         endif
      endif

      end

File: libraries/hds/cmpType.c
#include "sae_par.h"
#include "dat_par.h"
#include "hds.h"

void cmpType( HDSLoc *struc, const char *comp, char type[DAT__SZTYP+1],
              int *status ) {
/*
*+
*  Name:
*     cmpType

*  Purpose:
*     Component Type enquiry.

*  Synopsis:
*     void cmpType( HDSLoc *struc, const char *comp, char type[DAT__SZTYP+1],
*                   int *status )

*  Description:
*     A type enquiry is made for a structure component.

*  Parameters:
*     struc
*        A locator associated with a structured data object.
*     name
*        Expression specifying the component name of a primitive
*        object contained in the structure.
*     type
*        Variable to receive the data type.
*     status
*        Variable holding the status value. If this variable is not
*        SAI__OK on input, the routine will return without action.
*        If the routine fails to complete, this variable will be
*        set to an appropriate error number.

*  Authors:
*     DSB: David S. Berry (EAO)

*  History:
*     03-AUG-2017 (DSB):
*        Original version, based on equivalent Fortran routine.

*-
*/

   HDSLoc *tloc = NULL;
   datFind( struc, comp, &tloc, status );
   datType( tloc, type, status );
   datAnnul( &tloc, status );

}


